src/interface/Update.cpp

```diff
@@ -114,10 +108,10 @@ void UpdateContainer(Container<Real> &in, Container<Real> &dudt_cont,
   int nvars = cout_iter.vars.size();
 
   for (int n = 0; n < nvars; n++) {
```

Comment:
This kernel is potentially going to hinder performance on the GPUs. We need to be able to run this in a single kernel like in K-Athena, putting `par_for` on the outermost loop, instead of having one kernel for each variable. We would also not want to have `Variable` objects within the kernel, we want to access clean `Kokkos:View`s or `ParArrayND`. Looping over an array of `Variable`s or `ParArrayND`s is probably also not going to be viable for performance.

Like in K-Athena, we would like all the variables in a mesh block to be in one contiguous data object that can be looped over.

Reply:
I've been meaning to add an issue for a redesigned ContainerIterator that is GPU friendly.  I'll add it.  This is certainly beyond the scope of the PR, though.